packages/language-support/src/syntaxValidation/syntaxValidation.ts

```diff
@@ -62,6 +66,35 @@ function detectNonDeclaredLabel(
   return undefined;
 }
 
+function generateSyntaxDiagnostic(
+  parsedProcOrFunc: ParsedProcedure,
+  severity: DiagnosticSeverity,
+  message: string,
+  deprecation: boolean = false,
+): SyntaxDiagnostic {
+  const rawText = parsedProcOrFunc.rawText;
+  const nameChunks = rawText.split('\n');
+  const linesOffset = nameChunks.length - 1;
+  const lineIndex = parsedProcOrFunc.line - 1;
+  const startColumn = parsedProcOrFunc.column;
+  const endColumn =
+    linesOffset == 0
+      ? startColumn + rawText.length
+      : nameChunks.at(-1)?.length ?? 0;
```

Comment:
This part is still used in `detectNonDeclaredLabel`, so maybe worth collapsing that here as well